src/components/PatchesPage/index.tsx

```diff
@@ -0,0 +1,168 @@
+import React from "react";
+import { ApolloError } from "@apollo/client";
+import styled from "@emotion/styled";
+import Checkbox from "@leafygreen-ui/checkbox";
+import Icon from "@leafygreen-ui/icon";
+import queryString from "query-string";
+import { useLocation, useHistory } from "react-router-dom";
+import { Analytics } from "analytics/addPageAction";
+import { Banners } from "components/Banners";
+import { PageSizeSelector } from "components/PageSizeSelector";
+import { Pagination } from "components/Pagination";
+import {
+  PageWrapper,
+  StyledInput,
+  FiltersWrapper,
+  PageTitle,
+} from "components/styles";
+import {
+  useBannerDispatchContext,
+  useBannerStateContext,
+} from "context/banners";
+import {
+  UserPatchesQueryVariables,
+  PatchesPagePatchesFragment,
+} from "gql/generated/types";
+import { withBannersContext } from "hoc/withBannersContext";
+import { useFilterInputChangeHandler, usePageTitle } from "hooks";
+import { MyPatchesQueryParams, ALL_PATCH_STATUS } from "types/patch";
+import { getPageFromSearch, getLimitFromSearch } from "utils/url";
+import { ListArea } from "./ListArea";
+import { StatusSelector } from "./StatusSelector";
+
+interface Props {
+  analyticsObject?: Analytics<
+    | { name: "Filter Patches"; filterBy: string }
+    | { name: "Filter Commit Queue" }
+    | { name: "Change Page Size" }
+  >;
+  pageTitle: string;
+  patches?: PatchesPagePatchesFragment;
+  error?: ApolloError;
+}
+
+const PatchesPageCore: React.FC<Props> = ({
+  analyticsObject,
+  pageTitle,
+  patches,
+  error,
+}) => {
+  const bannersState = useBannerStateContext();
+  const dispatchBanner = useBannerDispatchContext();
+  const { replace } = useHistory();
+  const { search, pathname } = useLocation();
+  const { limit, page, includeCommitQueue } = getPatchesInputFromURLSearch(
+    search
+  );
+  const [
+    patchNameFilterValue,
+    patchNameFilterValueOnChange,
+  ] = useFilterInputChangeHandler(
+    MyPatchesQueryParams.PatchName,
+    false,
+    (filterBy: string) =>
+      analyticsObject?.sendEvent({ name: "Filter Patches", filterBy })
+  );
+  usePageTitle(pageTitle);
+  const onCheckboxChange = (): void => {
+    replace(
+      `${pathname}?${queryString.stringify(
+        {
+          ...queryString.parse(search, { arrayFormat }),
+          [MyPatchesQueryParams.CommitQueue]: !includeCommitQueue,
+        },
+        { arrayFormat }
+      )}`
+    );
+    // eslint-disable-next-line no-unused-expressions
+    analyticsObject?.sendEvent({ name: "Filter Commit Queue" });
```

Comment:
Same comment as above this should be called through the method supplied by the useUserpatchesAnalytics or a similar hook